src/EventsHeader/EventsHeader.jsx

```diff
@@ -0,0 +1,262 @@
+import css from './EventsHeader.module.css';
+
+// https://material.io/tools/icons/?style=baseline
+import {
+	Edit as EditIcon,
+	Event as EventIcon,
+	Movie as MovieIcon,
+	Portrait as PortraitIcon,
+} from '@material-ui/icons';
+import { purple } from '@material-ui/core/colors';
+
+const styles = theme => ({
+	button: {
+		color: theme.palette.getContrastText(purple[500]),
+		backgroundColor: purple[500],
+		'&:hover': {
+			color: 'white',
+			backgroundColor: purple[700],
+		},
+	},
+});
+
+const navButtons = [
+	{
+		icon: <EventIcon />,
+		label: 'events',
+		to: 'previous-events',
+	},
+	{
+		icon: <EditIcon />,
+		label: 'blog',
+		href: 'https://medium.com/codestar-blog',
+	},
+	{
+		icon: <PortraitIcon />,
+		label: 'pics',
+		to: 'previous-events',
+	},
+	{
+		icon: <MovieIcon />,
+		label: 'video',
+		href: 'https://www.youtube.com/channel/UCqwHhJNEUe7D-HGsX4zvKzQ',
+	},
+];
+
+@translate(['events'], { wait: true })
+export class EventsHeader extends Component {
+	constructor(props) {
+		super(props);
+		this.renderDetailsSection = this.renderDetailsSection.bind(this);
+	}
+
+	renderNavButtons() {
+		return navButtons.map(config => {
+			return (
+				<EventsHeaderButton
+					key={config.label}
+					label={config.label}
+					icon={config.icon}
+					to={config.to}
+					href={config.href}
+				/>
+			);
+		});
+	}
+
+	renderHeaderContent(mEvent, formattedDate) {
+		const { t, classes } = this.props;
+		return (
+			<Container fullHeight center>
+				<div className="row">
+					<div className="col-12">
+						<div className="mt-4">
+							<Typography
+								align="center"
+								variant="display1"
+								style={{ color: 'white' }}
+							>
+								{t('OUR_NEXT_EVENT')}
+							</Typography>
+							<Hidden mdUp>
+								<Typography
+									align="center"
+									variant="display2"
+									className={css.nextEventTitle}
+								>
+									{mEvent.name}
+								</Typography>
+							</Hidden>
+							<Hidden smDown>
+								<Typography
+									align="center"
+									variant="display4"
+									className={css.nextEventTitle}
+								>
+									{mEvent.name}
+									{/*TODO: buttons for the rest of the page, re-use style-color-white */}
+								</Typography>
+							</Hidden>
+							<Typography
+								gutterBottom
+								align="center"
+								variant="display2"
+								style={{ color: 'white' }}
+							>
+								{formattedDate}
+							</Typography>
+							<div style={{ textAlign: 'center' }}>
+								<Button
+									color="primary"
+									variant="raised"
+									href={mEvent.link}
+									className={`mr-1 ${classes.button}`}
+								>
+									{t('SIGN_UP')}
+								</Button>
+								<Link to="event-details" hashSpy smooth>
+									<Button variant="contained">{t('MORE_INFO')}</Button>
+								</Link>
+								<div className="mt-5">{this.renderNavButtons()}</div>
+							</div>
+						</div>
+					</div>
+				</div>
+			</Container>
+		);
+	}
+
+	renderNoEvents() {
```

Comment:
Consider making this a component as well. Something like `<EventsFeedback text={t('INFO_NO_NEXT_EVENTS')} />`? Then you can reuse the layout for other messages as well!

Reply:
There is no reason why the message that is styled this way would have to be re-used elsewhere, but I do dislike that the file is too big. I will split it into a sub component for EventsHeader.